common/web/gesture-recognizer/src/test/auto/headless/trackedPath.js

```diff
@@ -318,4 +321,87 @@ describe("TrackedPath", function() {
       assertSegmentSimilarity(reproedSegments[5], originalSegments[5], 'hold');  // ~300ms
     });
   });
+
+  describe('.segments event/Promise ordering', function() {
+    beforeEach(function() {
+      this.fakeClock = sinon.useFakeTimers();
+    })
+
+    afterEach(function() {
+      // NOTE:  for debugging investigations, it may be necessary to use .only on
+      // the test under investigation and to disable the `this.fakeClock.restore()` line.
+      //
+      // Tests tend to timeout when interactively debugging, and having unmocked timers
+      // suddenly restored during investigation can cause some very confusing behavior.
+      this.fakeClock.restore();
+    });
+
+    it('synthetic sequence: start -> hold -> move -> end', async function() {
```

Comment:
This (new) unit test is derived from a first-pass extreme rough-draft/prototype for gesture synthesis control flow handling; its needs motivated most of the other changes in this PR.